docs: Add example for Google Tag Manager's consent mode

silenzium opened the pull request:
The script in this example will result in a fine-grained consent experience for the user.

The following services can be toggled separately:

- ad_storage
- ad_user_data
- ad_personalization
- analytics_storage
- functionality_storage 
- personalization_storage
- security_storage

Preview: https://stackblitz.com/edit/stackblitz-starters-pybz8y?file=script.js

Closes #669
Related to #110

orkun1675 commented:
When testing with Google Tag Assistant I noticed two issues with this example:

1. I believe we need the brackets below when setting default consent to denied:

```javascript
// Set default consent to 'denied'
gtag('consent', 'default', {
    [SERVICE_AD_STORAGE]: 'denied',
    [SERVICE_AD_USER_DATA]: 'denied',
    [SERVICE_AD_PERSONALIZATION]: 'denied',
    [SERVICE_ANALYTICS_STORAGE]: 'denied',
    [SERVICE_FUNCTIONALITY_STORAGE]: 'denied',
    [SERVICE_PERSONALIZATION_STORAGE]: 'denied',
});
```

2. I got the error "A tag read consent state before a default was set". Do you think the snippet above has to be executed before the following pre-existing code I had in the header?

```javascript
gtag('js', new Date());	
gtag('config', '{{ site.google_analytics }}');
```

silenzium commented:
@orkun1675 thx, you are totally right, fixed it.
And point 2 makes sense, try to set defaults before any other dataLayer update. I do this and don't get any errors like you have (also updated the docs about it)

orkun1675 commented:
This is working for me now by the way. We are using this on a live website.

orestbida commented:
Thank you for your contribution!

alxndr-w commented:
@silenzium thank you for making this accessible.
